Sources/CardPayments/APIRequests/ConfirmPaymentSourceRequest.swift

```diff
@@ -36,7 +36,11 @@ struct ConfirmPaymentSourceRequest: APIRequest {
         path = String(format: pathFormat, orderID)
         
         jsonEncoder.keyEncodingStrategy = .convertToSnakeCase
-        body = try jsonEncoder.encode(confirmPaymentSource)
+        do {
+            body = try jsonEncoder.encode(confirmPaymentSource)
+        } catch {
+            throw CardClientError.encodingError
+        }
```

Comment:
Are we able to add a unit test for this logic?

Reply:
No because I added this general block as a safety in case someone forgets in the future. Besides the unlikely encoding error case that I handled on this ticket, I believe we are catching all errors. I found error throwing difficult to track, coming from many different sources. If some error is thrown without being cast as CoreSDKError, without that general catch block, the app will either crash (without that block at all) or give no indication of anything happening (nothing in catch block)

Reply:
I just added one but I don't think it's worth adding extra code in` ConfirmPaymentSourceRequest` for encoding error unit test

Reply:
addressed in commit 4de094c2c79d020a4e248ad4c5677b8d55c02d2e